src/block/block_compact.c

```diff
@@ -86,23 +286,35 @@ __wt_block_compact_progress(WT_SESSION_IMPL *session, WT_BLOCK *block, u_int *ms
     time_diff = WT_TIMEDIFF_SEC(cur_time, session->compact->begin);
     if (time_diff / WT_PROGRESS_MSG_PERIOD > *msg_countp) {
         ++*msg_countp;
-        __wt_verbose_debug1(session, WT_VERB_COMPACT_PROGRESS,
-          " compacting %s for %" PRIu64 " seconds; reviewed %" PRIu64 " pages, rewritten %" PRIu64
-          " pages",
-          block->name, time_diff, block->compact_pages_reviewed, block->compact_pages_rewritten);
+
+        if (block->compact_pages_rewritten_expected == 0)
+            __block_compact_estimate_remaining_work(session, block);
+
+        /*
+         * If we don't have the estimate at this point, it means that there wasn't enough work done
+         * to justify having an estimate, which should be extremely rare at this point. So it should
+         * be okay for us to skip the message in that case.
```

Comment:
I'm not an expert on usability, but it seems that it might still be useful to have a message just saying how many pages have been moved and saying something about still estimating total runtime. Depending on system load and how much time we spend walking the BTree to find the required thousand pages, it might take long enough that a user would want to see some message.

Or, an alternate thought, maybe we could get that thousand samples faster by computing the average size of the first thousand pages we look at, regardless of whether we rewrite them?